frontend/src/main/scala/bloop/engine/State.scala

```diff
@@ -97,4 +96,22 @@ object State {
     ExecutionContext.executor.setCorePoolSize(threads)
     state
   }
+
+  import bloop.Project
+  import bloop.io.AbsolutePath
+  def loadStateFor(
+      configDirectory: AbsolutePath,
+      commonOptions: CommonOptions,
+      logger: Logger
+  ): State = {
+    State.stateCache.getStateFor(configDirectory) match {
```

Comment:
Shouldn't we just use `computeIfAbsent`?

Reply:
Actually, `addIfMissing` relies on `computeIfAbsent`, so I can just use that.